src/main/java/co/uk/cloudam/streaming/template/discord/link/LinkCommand.java

```diff
@@ -0,0 +1,79 @@
+package co.uk.cloudam.streaming.template.discord.link;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.time.LocalDateTime;
+import java.util.Optional;
+import java.util.UUID;
+
+import co.uk.cloudam.streaming.template.discord.SlashCommand;
+import discord4j.core.event.domain.interaction.ChatInputInteractionEvent;
+import lombok.extern.apachecommons.CommonsLog;
+import reactor.core.publisher.Mono;
+
+@Component
+@CommonsLog
+public class LinkCommand implements SlashCommand {
+
+    private final LinkCodeRepo linkCodeRepo;
+
+    @Autowired
+    public LinkCommand(LinkCodeRepo linkCodeRepo) {
+        this.linkCodeRepo = linkCodeRepo;
+    }
+
+    @Override
+    public String getName() {
+        return "link";
+    }
+
+    @Override
+    public Mono<Void> handle(ChatInputInteractionEvent event) {
+        var newToken = UUID.randomUUID();
+        var discordName = event.getInteraction().getUser().getUsername();
+        if (doNew(discordName)) {
```

Comment:
This can be cleaned up just a little more.
At the moment you are getting from the DB 2 times. 

perhaps you can use the optional and pass it around in some method parameters?

Reply:
I think? .. I see what you mean. Fixed.